contributors/design-proposals/kubectl-create-configmap-env-file.md

```diff
@@ -0,0 +1,55 @@
+# Kubectl create configmap --env-file 
+
+## Goals
+
+`kubectl create configmap` allows a user to create configmaps via the command
+line. There should be easier ways to create a `ConfigMap`. One such way is to
+allow a Docker environment file (.env) to populate a `ConfigMap`.
+
+## Design
+
+The `create configmap` subcommand would add a new option called
+`--from-env-file`. The option will accept a single file. The option may not be
+used in conjunction with `--from-file` or `--from-literal-value`.
+
+### Environment file specification
+
+An environment file consists of lines to be in VAR=VAL format. Lines beginning
+with # (i.e. comments) are ignored, as are blank lines. Any whitespace in
+front of the VAR is removed. VAR must be a valid C_IDENTIFIER.  If the line
+consists of just VAR, then the VAL will be given a value from the current
+environment.
```

Comment:
Actually there is something I just realized. Because of the optional prefix, the keys can be invalid and once combined with the prefix can be made valid. This means I am not certain if requiring the keys is valid.